dialog/file_test.go

```diff
@@ -57,6 +59,92 @@ func TestEffectiveStartingDir(t *testing.T) {
 
 }
 
+func TestFileDialogResize(t *testing.T) {
+	win := test.NewWindow(widget.NewLabel("Content"))
+	win.Resize(fyne.NewSize(600, 400))
+	file := NewFileOpen(func(file fyne.URIReadCloser, err error) {
+	}, win)
+	file.SetFilter(storage.NewExtensionFileFilter([]string{".png"}))
+
+	//Mimic the fileopen dialog
+	d := &fileDialog{file: file}
+	open := widget.NewButton("open", func() {})
+	ui := fyne.NewContainerWithLayout(layout.NewBorderLayout(nil, nil, nil, open), open)
+	ui.Resize(fyne.NewSize(300, 280))
+	originalSize := ui.MinSize().Add(fyne.NewSize(fileIconCellWidth*2+theme.Padding()*4,
+		(fileIconSize+fileTextSize)+theme.Padding()*4))
+	d.win = widget.NewModalPopUp(ui, file.parent.Canvas())
+	d.win.Resize(originalSize)
+
+	file.dialog = d
+
+	//Test resize - bigger size
+	size := fyne.NewSize(800, 600)
+	file.Resize(size)
+	maxSize := file.dialog.win.Size()
+	minSize := file.dialog.win.MinSize()
+	//Test width
+	expectedWidth := size.Width
+	if size.Width > maxSize.Width {
+		expectedWidth = maxSize.Width
```

Comment:
A test case should set things up and they verify against known values.
This should be checking that the dialog is the expected size, the bounds checking should not be needed.